src/com/zebrunner/jenkins/pipeline/Repository.groovy

```diff
@@ -43,6 +49,30 @@ class Repository extends BaseObject {
         logger.info("Repository->register")
         Configuration.set("GITHUB_ORGANIZATION", Configuration.get(SCM_ORG))
         Configuration.set("GITHUB_HOST", Configuration.get(SCM_HOST))
+
+        this.scmHost = Configuration.get(SCM_HOST)
+        this.scmOrg = Configuration.get(SCM_ORG)
+        this.repo = Configuration.get(REPO)
+        this.branch = Configuration.get(BRANCH)
+        this.scmUser = Configuration.get(SCM_USER)
+        this.scmToken = Configuration.get(SCM_TOKEN)
+
+        logger.debug("scmHost: $scmHost scmOrg: $scmOrg repo: $repo branch: $branch")
+
+        switch (scmHost) {
+            case ~/^.*github.*$/:
+                this.scmClient = new GitHub(context, scmHost, scmOrg, repo, branch)
+                this.scmWebHookArgs = GitHub.getHookArgsAsMap(GitHub.HookArgs)
+                break
+            case ~/^.*gitlab.*$/:
+                this.scmClient = new Gitlab(context, scmHost, scmOrg, repo, branch)
+                this.scmWebHookArgs = Gitlab.getHookArgsAsMap(Gitlab.HookArgs)
+                break
+            case ~/^.*bitbucket.*$/:
+                this.scmClient = new BitBucket(context, scmHost, scmOrg, repo, branch)
+                this.scmWebHookArgs = BitBucket.getHookArgsAsMap(BitBucket.HookArgs)
+                break
+        }
```

Comment:
why we don't have default? I suspect as minimal raising exception in such case...

Reply:
Will add that right now, any other scm type we should support ?